proposals/text/5289-recency-reranker.md

```diff
@@ -0,0 +1,57 @@
+- Title: Addition of a 'RecencyReranker' node
+- Decision driver: @elundaeva
+- Start Date: 2023-07-05
+- Proposal PR: https://github.com/deepset-ai/haystack/pull/5289
+- Github Issue or Discussion: some links available in the "Detailed design" section below
+
+# Summary
+
+This reranker allows to have retrieved documents sorted not only by relevance (default) but also with recency factored in.
+
+# Basic example
+
+The RecentnessReranker would be instantiated as follows:
+
+  ``` python
+
+	  reranker = RecentnessReranker(
+	      date_identifier="date",
+	      weight="0.5",
+          top_k=3,
+	      method="reciprocal_rank_fusion",
+	  )
+  ```
+
+# Motivation
+
+Initially this reranker was implemented by Timo for a customer case where the date of the document mattered for retrieval. The reason we would like to add it to Haystack is because we see wider use for this node in future customer cases. One example where document recency matters is in a QA solution based on technical documentation with release notes of a software product - the older release notes should naturally have less priority in the responses than the most recent ones. 
```

Comment:
Great motivation! I would also expand the language in this to include the OSS community in addition to customer use cases. Additionally, I think adding other examples such as News Paper articles would help explain why we see a wider use of this node.